### DISCUSSION TITLE
Ability to output to a file with colours included?

### DISCUSSION THREAD

**Discussion by tom0010** [dandavison/delta#1338] [Q&A]
As the title says really, is this possible to do? 
I tried redirecting to a file `> output.txt` but because delta uses colours it doesn't render properly. 
Maybe the possibility to render to a pdf?

**Comment by dandavison** [comment]
Hey @tom0010. Delta produces colours as ANSI sequences that are understood by terminal emulators. I think the best way to do this would be to convert the ANSI sequences to HTML color codes and open the resulting file in a web browser, and then e.g. print to PDF. I just got that to work; you might want to play with the options a bit (`ansifilter` can't handle things like decoration boxes and hyperlinks so the options I added are disabling them):

First install `ansifilter` (for me on MacOS that is `brew install ansifilter`). Then

```
git show \
    | delta --no-gitconfig --file-decoration-style blue --hunk-header-decoration-style blue \
    | ansifilter --html \
    > /tmp/diff.html
```

Now open /tmp/diff.html in a web browser:


<table>
    <tbody>
        <tr>
            <td><img width="400" alt="image" src="https://user-images.githubusercontent.com/52205/224020893-ae1ceee4-1f58-4b4b-aece-a5c17e80c98f.png"></td>
        </tr>
    </tbody>
  </table>

**Reply by tom0010** [reply]
Thanks @dandavison, where this does give me a html page with the diff. The colours are all messed up, here is an example attached.

I seem to be able to get around this when I pass `--light`. Guessing because my default is a dark terminal. 

![image](https://user-images.githubusercontent.com/41154665/227171435-c4b5557f-bf5b-4bfe-9648-7490a39458de.png)

**Reply by dandavison** [reply]
OK thanks, yeah I guess I have an env var set telling delta to use a light theme, despite the fact I supplied `--no-gitconfig`. And that happened to do the right thing because for display in a web browser that is going to be a light background by default!

**Reply by tom0010** [reply]
Just coming back to this, I normally use `--side-by-side` when we use the ansifilter, it adds extra special chars, so the HTML doesn't really look that great.

**Reply by tom0010** [reply]
Furthermore, the HTML page that it generates is double the size of a vimdiff to html. 
I am wondering if we could pass `delta` into vimdiff somehow?

Here's the command I'm using to generate a vimdiff:
`vimdiff  file1.cfg file2.cfg -c TOhtml -c 'w! diff.html' -c 'qa!'`

Delta looks a lot more appealing than vimdiff, I've tried multiple colour schemes with vim and none seem to be that great, I ended up landing on gruvbox in dark mode, which seems ok. 

@dandavison
